packages/bot/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -180,14 +180,14 @@ Blockly.Blocks.trade_definition_tradeoptions = {
                 if (target_block && target_block.isShadow()) {
                     const min_duration = durations.find(d => d.unit === this.selected_duration);
 
-                    if (min_duration) {
+                    if (min_duration && event_group !== 'load') {
                         target_block.setFieldValue(min_duration.min, 'NUM');
                     }
                 }
             }
         });
     },
-    updateBarrierInputs(should_use_default_types) {
+    updateBarrierInputs(should_use_default_types, event_group) {
```

Comment:
is there any reason why we have to pass in event group each of every functions?, can it be place on store or something that can be accessed anywhere?

Reply:
It will be more troublesome because we need to update `event` on every Blocks `onchange` function.